api/services/Reddit.js

```diff
@@ -65,6 +68,18 @@ var makeRequest = async function (refreshToken, requestType, url, data, rateLimi
   return bodyJson;
 };
 
+var getEntireListing = async function (refreshToken, endpoint, query, rateThreshold, after, before) {
+  var url = endpoint + query + (query ? '&' : '?') + 'count=102&limit=100' + (after ? '&after=' + after : '') + (before ? '&before=' + before : '');
+  var batch = await makeRequest(refreshToken, 'GET', url, undefined, rateThreshold, after, before);
+  var results = batch.data.children;
+  if ((after || after === undefined) && batch.data.after === null || before && batch.data.before === null) {
```

Comment:
I think there might be something wrong with this if statement?

If after is truthy, or it's undefined? 

Are the brackets wrong? Run throguh the logic with me.